### ISSUE TITLE
Why masked out EOS in SFT training?

### ISSUE THREAD

**Issue by wenxueru** [volcengine/verl#655]
In the `verl.utils.sft_dataset.py` file, why does the `__getitem__` method of `SFTDataset` class (around line 167) mask out the EOS token? If we do this when training a pretrained model, could it potentially lead to never-ending generation?

**Comment by YinongLong** [comment]
> In the `verl.utils.sft_dataset.py` file, why does the `__getitem__` method of `SFTDataset` class (around line 167) mask out the EOS token? If we do this when training a pretrained model, could it potentially lead to never-ending generation?

You can understand the ​`loss_mask` here as masking the parts where the loss calculation isn't required when predicting the next token. Masking the EOS is reasonable because the EOS token's prediction doesn't need to be considered in loss calculation. In practice, this `loss_mask` needs to be understood in conjunction with the loss calculation logic in the training section . For simplicity, you can check line 313 in file ​`fsdp_sft_trainer.py` to assist your understanding.
